auto-value-gson/src/main/java/com/ryanharter/auto/value/gson/AutoValueGsonExtension.java

```diff
@@ -527,16 +537,29 @@ public TypeSpec createTypeAdapter(Context context, ClassName className, ClassNam
   public List<MethodSpec> createDefaultMethods(ClassName gsonTypeAdapterName, List<Property> properties) {
     List<MethodSpec> methodSpecs = new ArrayList<>(properties.size());
     for (Property prop : properties) {
-      ParameterSpec valueParam = ParameterSpec.builder(prop.type, "default" + upperCamelizeHumanName(prop)).build();
```

Comment:
When using AutoValue with builders and optional properties, the builder setters can take the wrapped type. In that vein, the setters are generated accepting the wrapped type. Note that an `Optional<Integer>` property would yield a setter that takes an `Integer` boxed type, whereas `OptionalInt` would yield a setter that takes a `int` primitive.

The generated setter code allows setters to accept null by using the `ofNullable` static method before assigning to the type adapter field.